### PR TITLE
Add pull_request_template.md

### LINKED ISSUE #11: Add pull_request_template.md
Add a pull request (PR) template to the ./github directory so contributors can automatically see the expected contents for the project in a new PR body.

The PR template will be based on these linked [examples](https://github.com/axolo-co/pull_request_template) provided by @axolo-co but modified to better suit this project.

### PR DESCRIPTION
**Pull request by R055A**
# Description

A pull request (PR) template added to the ./github directory so contributors can automatically see the expected contents for the project in a new PR body.

The PR template is based on these linked [examples](https://github.com/axolo-co/pull_request_template) provided by @axolo-co but modified to better suit this project.

#### Template

This is what is seen when editing a new PR:

```
<!-- Is there a relevant and understandable title succinctly summarizing the changes? -->

# Description

<!--- Give a description of the changes here in more detail -->
<!--- Include screenshots if necessary -->

## Related Issue

<!--- This project only accepts pull requests related to already existing open issues -->
<!--- Please link to the open issue here: -->

Solves #<!--issue-->

<!--Don't forget to move the related issue to the 'Review in progress' column.-->

## Type of change

<!--Please delete options that are not relevant.-->

- [ ] New feature (enhancement)
- [ ] Bug fix
- [ ] Refactor
- [ ] Breaking change
- [ ] Documentation

<!--Don't forget to add a label to the pull request, if relevant.-->

## How Has This Been Tested?

<!--- Please describe in detail how the changes have been tested. -->
<!--- Include details of the testing environment, the tests run, and the results. -->

<!--Please delete options that are not relevant.-->

- [ ] Automated testing
- [ ] Manual testing

# Checklist:

<!--Please delete options that are not relevant.-->

- [ ] Does a similar (open or closed) pull request not already exist?
- [ ] Is the pull request _head_ repository a fork repository?
- [ ] Is the pull request _compare_ branch a development branch?
- [ ] Is the code documented, particularly in hard-to-understand areas?
- [ ] Does the code build without new warnings?
- [ ] Has testing been performed that proves changes are effective and work?
- [ ] Has a self- and/or peer-review of the code been performed?
- [ ] Have dependent changes been merged and published in downstream modules?
- [ ] Does all new and existing automated testing pass?
- [ ] Is the person responsible for the repository assigned to the pull request?
- [ ] Is the pull request linked to a project?
- [ ] Is the pull request linked to a milestone?

For more information, refer to the [Contributing Guidelines](../blob/contribution-guidelines/.github/CONTRIBUTING.md) document.

<!--Don't forget to notify others of the pull request for a quick review.-->
```

## Related Issue

Solves #11 

## Type of change

- [x] Documentation

# Checklist:

- [x] Does a similar (open or closed) pull request not already exist?
- [x] Is the pull request _head_ repository a fork repository?
- [x] Is the pull request _compare_ branch a development branch?
- [x] Has a self- and/or peer-review of the code been performed?
- [x] Does all new and existing automated testing pass?
- [x] Is the person responsible for the repository assigned to the pull request?
- [x] Is the pull request linked to a project?
- [x] Is the pull request linked to a milestone?

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+- [ ] Is the code documented, particularly in hard-to-understand areas?
+- [ ] Does the code build without new warnings?
+- [ ] Has testing been performed that proves changes are effective and work?
+- [ ] Has a self- and/or peer-review of the code been performed?
+- [ ] Have dependent changes been merged and published in downstream modules?
+- [ ] Does all new and existing automated testing pass?
+- [ ] Is the person responsible for the repository assigned to the pull request?
+- [ ] Is the pull request linked to a project?
+- [ ] Is the pull request linked to a milestone?
+
```
Comment (Xiaoqi510) [inline .github/pull_request_template.md:55]: # Screenshot

Original                   |Updated
:---------------------:|:------------------:
**original screenshot ** | ** updated screenshot **|

**Review by Xiaoqi510 — CHANGES_REQUESTED** [review]
We could also add a screenshot to show the difference between the original version and updated version code, that can help other reviewers easily understand the changes.

File: .github/pull_request_template.md (hunk 1 above)
Comment (r055a) [reply .github/pull_request_template.md:55]: Thank you for your review and feedback @Xiaoqi510 

Regarding using before and after screenshots for reviewing code changes, there is already the suggested option for adding screenshots in the template description and I don't think it is necessary to have screenshots as a requirement when we can already view code changes in the pull request commit(s).

Comment (Xiaoqi510) [reply .github/pull_request_template.md:55]: You are right! I forget the "files-changed" function of github :)

**Review by Xiaoqi510 — DISMISSED** [review]

**Comment by yuchendong13** [comment]
Hi, sorry just want to clarify with these two points:

- [ ] Is this pull request head a fork repository?
- [ ] Is this pull request compare a development branch?

I don't quite follow what these mean. Do these actions automatically generate a pull request or something?

**Comment by R055A** [comment]
> Hi, sorry just want to clarify with these two points:
> 
>     * [ ]  Is this pull request head a fork repository?
> 
>     * [ ]  Is this pull request compare a development branch?
> 
> 
> I don't quite follow what these mean. Do these actions automatically generate a pull request or something?

Thank you for the review @yuchendong13 

- The base repository is the repository requested to merge to, and the **head** repository is the repository the merge is from. 
- The base is the branch requested to merge to in the base repository, and the **compare** is the branch in the head repository the merge is from.

I hope that this answers your question. You will see when making a pull request, but I will make it more descriptive anyway.

**Review by hiin3d55 — APPROVED** [review]
LGTM!

**Review by Xiaoqi510 — APPROVED** [review]

**Review by hpar461 — APPROVED** [review]

**Review by AlexMiao7 — APPROVED** [review]

**Review by qzho236 — APPROVED** [review]

**Review by christopher-alba — APPROVED** [review]

**Pull request merged**